### PR TITLE
Update rules_apple and rules_swift

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

File: test/fixtures/generator/bwx_spec.json [hunk 1]
```diff
@@ -77,6 +77,7 @@
                 "DEBUG_INFORMATION_FORMAT": "dwarf",
                 "ENABLE_BITCODE": false,
                 "GCC_OPTIMIZATION_LEVEL": "0",
+                "PRODUCT_MODULE_NAME": "tools_generator_test_tests",
```
Comment (brentleyjones) [inline test/fixtures/generator/bwx_spec.json:80]: This because of the rules_swift change. Looking into it.

Comment (brentleyjones) [reply test/fixtures/generator/bwx_spec.json:80]: It's expected.

File: README.md [hunk 2]
```diff
@@ -52,8 +52,8 @@ rules.
 - macOS 12.0-12.4
 - Xcode 13.0-13.4.1
 - Bazel 5.2.0
```
Comment (jpsim) [inline README.md:54]:
```suggestion
- Bazel 5.2.0-5.3.0
```

**Review by jpsim — APPROVED** [review]
Would it be possible to set up a version matrix on CI to test the various supported combinations?

File: shared.bazelrc [hunk 3]
```diff
@@ -3,7 +3,7 @@ startup --output_base=bazel-output-base
 build --incompatible_disallow_empty_glob
 
 build --remote_default_exec_properties=OSFamily=darwin
-build --remote_default_exec_properties=cache_bust=macOS_12.0/2
+build --remote_default_exec_properties=cache_bust=macOS_12.0/3
```
Comment (brentleyjones) [inline shared.bazelrc:6]: I poisoned the cache having an override.

**Comment by brentleyjones** [comment]
> Would it be possible to set up a version matrix on CI to test the various supported combinations?

I'm looking into this. If not right now, I bet we can soon.

**Comment by brentleyjones** [comment]
https://github.com/buildbuddy-io/rules_xcodeproj/issues/247#issuecomment-1224750633

**Pull request merged**